### DISCUSSION TITLE
Unable to Serial Print and Serial Plot on Serial1 with STM32duino core ( Solved )

### DISCUSSION THREAD

**Discussion by TeguhTeknisi** [stm32duino/Arduino_Core_STM32#1862] [Q&A] [answered]
Hi,
Why i don't see anything on Serial Port 1 when using Serial Monitor or Serial Plotter when using STM32duino core?
when i using Roger core, i can see it.

Nevermind, its because i choose Board part number: Generic F103C6Tx 
When i change to BluePill F103C6 (32K) , i can see it.
I thought, i have Bluepill clone, so i choose Generic F103C6Tx 

Here the code i use to test
```
void setup() {
  Serial.begin(115200);

  pinMode(PC13, OUTPUT);
}

void loop() {
  Serial.println("Led Blynking....");
  digitalWrite(PC13, HIGH);
  delay(1000);
  digitalWrite(PC13, LOW);
  delay(1000);
}
```
I have try it using  STM32F103C6T6 board and STM32F103C8T6 board

STM32F103C6T6
```
Board       : --
Chip ID: 0x412 
BootLoader protocol version: 2.2
Device name : STM32F101/F102/F103 Low-density
Flash size  : 32 KBytes (default)
Device type : MCU
Revision ID : --  
Device CPU  : Cortex-M3
```

STM32F103C8T6
```
Board       : --
Chip ID: 0x410 
BootLoader protocol version: 2.2
Device name : STM32F101/F102/F103 Medium-density
Flash size  : 128 KBytes (default)
Device type : MCU
Revision ID : --  
Device CPU  : Cortex-M3
```

OS Kubuntu Linux
Arduino IDE 1.6.19
STM32duino core 2.3.0
Using USB Serial CH340 to PA9 and PA10 to upload the code

Best regrads,
Teguh

**Comment by fpistm** [comment, marked as answer]
Hi,
this is normal.
BluePill defined default `Serial `instance to be mapped on `Serial1 `(PA9/PA10):
https://github.com/stm32duino/Arduino_Core_STM32/blob/75b051e50cb2b02cb110d69433917e5c6d14ad96/variants/STM32F1xx/F103C4T_F103C6(T-U)/variant_BLUEPILL_F103C6.h#L125-L137
While Generic mapped `Serial `to `Serial2 `(PA3/PA2):
https://github.com/stm32duino/Arduino_Core_STM32/blob/75b051e50cb2b02cb110d69433917e5c6d14ad96/variants/STM32F1xx/F103C4T_F103C6(T-U)/variant_generic.h#L131-L143

See https://github.com/stm32duino/wiki/wiki/API#hardwareserial

> By default, only one Serialx instance is available mapped to the generic Serial name.

**Reply by TeguhTeknisi** [reply]
Thanks @fpistm 

Now i have to find out, why i cannot compile emonlib for stm32 ( https://github.com/openenergymonitor/STM32/issues/20 )

**Reply by fpistm** [reply]
> Thanks @fpistm
> 
> Now i have to find out, why i cannot compile emonlib for stm32 ( [openenergymonitor/STM32#20](https://github.com/openenergymonitor/STM32/issues/20) )

Looking quickly to this issue, your code seems wrong: `PA_0 `is type of PinName while Arduino deal with pin number. So you have to use `PA0` which represent the pin number. Same for `PA_1 `which sould be `PA1`
  ```patch
-  emon1.voltage(PA_0, 268.97, 1.7);  // Voltage: input pin, calibration, phase_shift
-  emon1.current(PA_1, 60.606);       // Current: input pin, calibration.
+  emon1.voltage(PA0, 268.97, 1.7);  // Voltage: input pin, calibration, phase_shift
+  emon1.current(PA1, 60.606);       // Current: input pin, calibration.

```
